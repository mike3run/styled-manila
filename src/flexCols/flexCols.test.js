import React from 'react'
import styled from 'styled-components'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { flexCols } from '..'

test('FlexCols with default base length', () => {
  const Test = styled.div`
    ${flexCols(6)}
  `
  const tree = renderer.create(<Test />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('flex-basis', '50%')
  expect(tree).toHaveStyleRule('max-width', '50%')
})

test('FlexCols with custom base length', () => {
  const Test = styled.div`
    ${flexCols(1, 4)}
  `
  const tree = renderer.create(<Test />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('flex-basis', '25%')
  expect(tree).toHaveStyleRule('max-width', '25%')
})
