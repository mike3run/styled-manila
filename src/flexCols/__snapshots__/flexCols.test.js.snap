// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexCols with custom base length 1`] = `
.c0 {
  max-width: 25%;
  -webkit-flex-basis: 25%;
  -ms-flex-preferred-size: 25%;
  flex-basis: 25%;
}

<div
  className="c0"
/>
`;

exports[`FlexCols with default base length 1`] = `
.c0 {
  max-width: 50%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
}

<div
  className="c0"
/>
`;
